vecvec-lapack/Vecvec/LAPACK/Internal/Compat.hs

```diff
@@ -33,3 +36,13 @@ updPtr f (ForeignPtr p c) = case f (Ptr p) of { Ptr q -> ForeignPtr q c }
 unsafeWithForeignPtr :: ForeignPtr a -> (Ptr a -> IO b) -> IO b
 unsafeWithForeignPtr = withForeignPtr
 #endif
+
+-- | Distance in elements between two pointers. Supplement for `advancePtr`.
+--
+distancePtr :: forall a . Storable a => Ptr a -> Ptr a -> Int
+-- distancePtr ptrFrom ptrTo = (ptrTo `minusPtr` ptrFrom) `div` (sizeOf (undefined :: a))
+distancePtr = distancePtr' undefined -- TODO WTF?? why ambigous??
+ where
+    distancePtr' :: Storable a => a -> Ptr a -> Ptr a -> Int
+    distancePtr' x ptrFrom ptrTo = (ptrTo `minusPtr` ptrFrom) `div` sizeOf x
```

Comment:
Without `ScopedTypeVariables` GHC will think that `a` is some unrelated type variable.

Reply:
Weird. I thought I tried `ScopedTypeVariables` :)